core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -1153,14 +1193,15 @@ class ReplicaManager(val config: KafkaConfig,
       logReadResultMap.put(topicIdPartition, logReadResult)
     }
 
-    // respond immediately if 1) fetch request does not want to wait
+    // Respond immediately if 1) no remote fetches are required
     //                        2) fetch request does not require any data
-    //                        3) has enough data to respond
-    //                        4) some error happens while reading data
-    //                        5) we found a diverging epoch
-    //                        6) has a preferred read replica
-    if (params.maxWaitMs <= 0 || fetchInfos.isEmpty || bytesReadable >= params.minBytes || errorReadingData ||
-      hasDivergingEpoch || hasPreferredReadReplica) {
+    //                        3) fetch request does not require any data
```

Comment:
`2)` should be fetch request does not want to wait. Also, the comment is written as if `1)` is also disjunctive, but it is not.